core/src/impl/Kokkos_DesulAtomicsConfig.hpp

```diff
@@ -17,13 +17,18 @@
 #ifndef KOKKOS_DESUL_ATOMICS_CONFIG_HPP
 #define KOKKOS_DESUL_ATOMICS_CONFIG_HPP
 
-#if defined(KOKKOS_ARCH_KEPLER) || defined(KOKKOS_ARCH_MAXWELL)
+#include <impl/Kokkos_NvidiaGpuArchitectures.hpp>
+
+#if KOKKOS_IMPL_ARCH_NVIDIA_GPU < 60
```

Comment:
We need to be careful when comparing w/o checking that the macro is defined.
I think we need
```suggestion
#if 60 > KOKKOS_IMPL_ARCH_NVIDIA_GPU  // DOES NOT WORK
```
**edit** see that is not even right...

But it might be a hint that this is too clever and we should come up with something less error prone, possibly guarding the whole thing with `#ifdef KOKKOS_IMPL_ARCH_NVIDIA_GPU`

Reply:
https://godbolt.org/z/Gnx1eE6Gd

Reply:
good catch. I think guarding the whole block is the easiest and most readable